include/zephyr/pm/pm.h

```diff
@@ -87,8 +156,13 @@ bool pm_state_force(uint8_t cpu, const struct pm_state_info *info);
  * list.
  *
  * @param notifier pm_notifier object to be registered.
+ * @param state PM state for which the notifier should be registered
+ * @param substate_id PM substate for which the notifier should be registered.
+ *
+ * @return 0 if the notifier was successfully registered, a negative value
+ * otherwise.
  */
-void pm_notifier_register(struct pm_notifier *notifier);
+int pm_notifier_register(struct pm_notifier *notifier, enum pm_state state, uint8_t substate_id);
 
```

Comment:
I am thinking about it, wouldn't be better having notifier only per state (and not state + substate). I wondering that including substate will make really hard re-using the driver across different SoCs. The callback though will include state / substate  .... or maybe one solution is to use the information about which states should do device pm (from  #60939) or something similar.  Obviously, that can be done later, that is why we can have a simple version of this pr in first.

Reply:
It is intended that the driver obtains the `power-state` directly from DTS via macros, so the driver will always have the correct state/substate combination. The state/substates combinations aren't intended to be hardecoded.

See my recommended implementation in the driver [here](https://github.com/zephyrproject-rtos/zephyr/pull/59200/commits/3931c014abfe45d0e93f6ab3999875e4185268d5#diff-54bdff4ff3a16e194201b66e24cada279fdfdd40dfd66b1df3df3a99d0f65cd0R2199-R2200).

Reply:
Agree with @knthm. @ceolin would you approve the current state then ?